src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -270,6 +274,8 @@ public sealed class ReadyToRunCodegenCompilation : Compilation
             _dependencyGraph.AddRoot(new Import(NodeFactory.EagerImports, instructionSetSupportSig), "Baseline instruction set support");
         }
 
+        private readonly static string s_folderUpPrefix = ".." + Path.DirectorySeparatorChar;
```

Comment:
Will this work on Unix? Should we choose `DirectorySeparatorChar` based on our target platform?

Reply:
Actually, I think you can disregard this. I was thinking about the relative path we embed in the non-composite images.

Reply:
The Path.DirectorySeparatorChar is platform specific.

Reply:
Yeah I was more worried about future cross-platform bug if we generate a Linux composite image from Windows. The change description indicates we burn relative paths in the component assemblies to the composite image but it seems like we actually just put the composite image name and look it up by simple name instead. If that's the case, then we should be fine.

Reply:
The first version of my change used reverse relative paths to go from the component assemblies to the composite native image. Based on David's PR feedback I simplified this logic by throwing away the relative path magic and just allowing explicit composite image path lookup which seems more general for additional reasons.

Reply:
Sigh, that's what I get for reviewing in a 2 day old browser tab 🙄  Thanks for clarifying